pepiscms/application/controllers/admin/Ajaxfilemanager.php

```diff
@@ -54,13 +54,15 @@ public function getjsonfilelist()
                 if ($file == '..' || $file == '.') {
                     continue;
                 }
-
+                $file_path = $user_files_dir . $current_path . $file;
                 $f['name'] = htmlspecialchars($file);
-                $f['mtime'] = filemtime($user_files_dir . $current_path . $file);
-                $f['ctime'] = filectime($user_files_dir . $current_path . $file);
-                $f['size'] = filesize($user_files_dir . $current_path . $file);
 
-                if (is_dir($user_files_dir . $current_path . $file)) {
+                if (is_readable($file_path)){
+                    $f['mtime'] = filemtime($file_path);
```

Comment:
Doesn't it cause problems when these indexes are not present?
Maybe you could consider populating them with some fallback values in the else branch so that we can have consistent structure of the files array?

Reply:
I've added array initialization